src/uu/sort/src/sort.rs

```diff
@@ -254,6 +254,7 @@ impl Output {
             let file = OpenOptions::new()
                 .write(true)
                 .create(true)
+                .truncate(true)
```

Comment:
seems that it regressed a test

Reply:
Oh, it was a problem that I didn't read the comments carefully, I'm very sorry!

Reply:
Here's the result on my computer after fixed

![image](https://github.com/uutils/coreutils/assets/20512501/f569cf27-a9e9-463d-ab1b-9b19e933b422)

Reply:
please don't share screenshot, they are terrible for accessibility and search